internal/server/data/migrations.go

```diff
@@ -1054,6 +1055,59 @@ func makeIdxEmailsProvidersUnique() *migrator.Migration {
 	}
 }
 
+func addDestinationCredentials() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2022-11-25T14:46",
+		Migrate: func(tx migrator.DB) error {
+			// Postgres 12 doesn't support CREATE OR REPLACE for triggers.
+			_, err := tx.Exec(`
+				CREATE TABLE IF NOT EXISTS destination_credentials (
+					id 							 	bigint NOT NULL,
+					organization_id  	bigint NOT NULL,
+					expires_at 				timestamp with time zone,
+					update_index 			bigint NOT NULL,
+					user_id 					bigint NOT NULL,
+					destination_id 		bigint NOT NULL,
+					answered          bool NOT NULL DEFAULT false,
+					--
+					bearer_token text
+				);
+
+				CREATE INDEX IF NOT EXISTS idx_cred_req_org_dest on destination_credentials (organization_id, destination_id);
+
+				CREATE OR REPLACE FUNCTION destination_credential_insert_notify() RETURNS trigger
+					LANGUAGE PLPGSQL
+					AS $$
+				BEGIN
+					-- on insert, we notify connector listeners for this destination
+					PERFORM pg_notify(current_schema() || '.credreq_' || uidinttostr(NEW.organization_id) || '_' || uidinttostr(NEW.destination_id), NEW.id::TEXT);
```

Comment:
The maximum postgres identifier length is 63 bytes: https://www.postgresql.org/docs/current/limits.html

Each of these ids can be up to 19 bytes when represented as a string (11 if we use the uid form).

I believe there are 10 other characters here to identify the channel and separate the ids.

That leaves 15 characters for the schema name if we used integer representation for the IDs, and 31 by using the uid representation.

That seems like a reasonable limit for the schema name.

Reply:
I hit this limit representing the integers as strings directly, thus the uid form.

Reply:
I had to add the schema for testing, without it tests would conflict with each other because these channels are global and not scoped to a schema.  It's possible we can find a different way to make unique names.